plugins/woocommerce/src/Admin/Features/Navigation/RemovedDeprecated.php

```diff
@@ -0,0 +1,42 @@
+<?php
+
+namespace Automattic\WooCommerce\Admin\Features\Navigation;
+
+use WC_Tracks;
+
+class RemovedDeprecated {
+	private static function handle_deprecated_method_call( $class, $name ) {
+		$logger = wc_get_logger();
+		
+		if ( $logger ) {
+			$logger->warning(
+				"$class is deprecated since 9.3 with no alternative. Please remove the call to $name."
+			);
+		}
+
+		if ( class_exists( 'WC_Tracks' ) ) {
+			WC_Tracks::record_event( 'deprecated_navigation_method_called' );
+		}
+	}
+
+	/**
+	 * Handle calls to deprecated methods.
+	 *
+	 * @param string $name The name of the deprecated method.
+	 * @param array $arguments The arguments passed to the deprecated method.
+	 */
+	public function __call( $name, $arguments ) {
+		self::handle_deprecated_method_call( get_called_class(), $name );
```

Comment:
This approach could log the original class name instead of the alias name.

```php
class RemovedDeprecated {
	/**
	 * The original class name for an alias class.
	 *
	 * @var string|null
	 */
	private static $originalClass;

	/**
	 * Set the original class name for an alias class.
	 *
	 * @param string $originalClass The original class name.
	 */
	public static function __constructStatic( $originalClass ) {
		self::$originalClass = $originalClass;
	}

	private static function handle_deprecated_method_call( $class, $name ) {
		$logger = wc_get_logger();

		if ( $logger ) {
			$logger->warning(
				self::$originalClass . " is deprecated since 9.3 with no alternative. Please remove the call to $name."
			);
		}

		if ( class_exists( 'WC_Tracks' ) ) {
			WC_Tracks::record_event( 'deprecated_navigation_method_called' );
		}
	}
```

```php
// When creating aliases
RemovedDeprecated::__constructStatic( \Automattic\WooCommerce\Admin\Features\Navigation\Menu::class );
class_alias( RemovedDeprecated::class, \Automattic\WooCommerce\Admin\Features\Navigation\Menu::class );

RemovedDeprecated::__constructStatic( \Automattic\WooCommerce\Admin\Features\Navigation\Screen::class );
class_alias( RemovedDeprecated::class, \Automattic\WooCommerce\Admin\Features\Navigation\Screen::class );

RemovedDeprecated::__constructStatic( \Automattic\WooCommerce\Admin\Features\Navigation\CoreMenu::class );
class_alias( RemovedDeprecated::class, \Automattic\WooCommerce\Admin\Features\Navigation\CoreMenu::class );

```

Reply:
I think `Navigation classes` is still a bit vague. It would be better to mention the specific namespace or class name.

How about `Navigation classes Automattic\WooCommerce\Admin\Features\Navigation\* are deprecated since 9.3 with no alternative...` 

or `The WooCommerce Admin Navigation feature and its classes (Screen, Menu, CoreMenu) are deprecated since 9.3 with no alternative...`?

Reply:
Thanks @chihsuan, fixed in https://github.com/woocommerce/woocommerce/pull/52248/commits/10a2a3d222b050430fdbc0263d3632e0625864d2